include/boost/hana/fwd/lazy.hpp

```diff
@@ -68,7 +68,11 @@ BOOST_HANA_NAMESPACE_BEGIN
     //! [1]: http://ldionne.com/2015/03/16/laziness-as-a-comonad
 #ifdef BOOST_HANA_DOXYGEN_INVOKED
     template <typename ...>
-    struct lazy;
+    struct lazy {
+        //! Equivalent to `hana::chain`.
+        template <typename ...T, typename F>
+        constexpr auto operator|(lazy<T...>, F);
```

Comment:
This should probably be made a `friend`, otherwise it does not make sense to receive two arguments. The same applies for some operators in the other files you modified.